intel-sgx/enclave-runner/src/tcs.rs

```diff
@@ -163,8 +170,8 @@ pub(crate) fn coenter<T: Tcs>(
                 options(att_syntax)
             );
             if ret == 0 {
-                sgx_result = run.function;
-                match sgx_result.try_into() {
+                enclu_leaf = run.function;
+                match enclu_leaf.try_into() {
                     Ok(Enclu::EExit) => { /* normal case */ },
```

Comment:
This is no longer the "normal case". When you interrupt the enclave in the signal handler, it explicitly writes `$EEXIT` to `enclu_leaf`, even though it should be `ERESUME`. See the [vdso](https://github.com/torvalds/linux/blob/dad9774deaf1cf8e8f7483310dfb2690310193d2/arch/x86/entry/vdso/vsgx.S#L70). You need to detect such cases and reenter the enclave. You can check the vdso to see which registers you can use for this.

Reply:
added EXITING flag which is tested in the signup handler addresses this issue as well